pbf/models.py

```diff
@@ -231,12 +231,29 @@ class Meta:
     #
     # To see the list of spirits that use this field, see spirit_specific_resource_name below.
     spirit_specific_resource = models.IntegerField(default=0)
+    # some spirits allow you to do something to their spirit-specific resource,
+    # but only once per turn.
+    # again, to reduce some repetitive code, we'll use one field for all of these.
+    spirit_specific_per_turn_flags = models.PositiveIntegerField(default=0)
+
+    # spirits that use spirit-specific per-turn flags:
+    # Spreading Rot Renews the Earth:
```

Comment:
notes; potential other uses I'm thinking of:

* whether Dances Up Earthquakes has added energy to impending cards this turn (discussed in https://github.com/nathanj/spirit-island-pbp/issues/52)
* whether Unconstrained Sharp Fangs Behind the Leaves has added a beast this turn (wouldn't block them from continuing to make adjustments, but just as a memory aid for whether it's already been done this turn)

Reply:
Thanks for putting this comment here; I probably would've come to that realization on further review (since I had seen the impending suggestion), but definitely was thinking where else this would be used on first glance last night.

I haven't played Unconstrained before; but, I do even feel like this could be used as a restriction:
- just like Rot, you have 2 flags (one for preparing, one for removing)
- if you do either, you can no longer prepare that turn
- if you remove, you can continue to remove that turn

Reply:
The reason I wanted to still allow changes in the other direction is just in case of misclicks or something, but I'm not sure how likely it is to come up. To be determined?

This idea you're presenting for the flags for Unconstrained Fangs is interesting! Note that since Unconstrained Fangs is using just the +1 -1 buttons, which call `change_spirit_specific_resource` . I think I'll just have two bits in there that will be generically SPIRIT_SPECIFIC_INCREMENTED_THIS_TURN and SPIRIT_SPECIFIC_DECREMENTED_THIS_TURN or something. I will update this shortly.

Reply:
Yup, I was thinking of two bits on the flag also, sorry for not specifying.

But, good point...I had in mind that this would be similar to energy gain, but of course energy gain always has a way to manually correct (which is needed for other use cases). It would suck that if you subtracted too many, there would be no way to fix it (unless you ready/unready multiple times in a turn).

The only option that I think is feasible if we want it more than an indicator: provide a free form input box to change the number of prepared beats in case you mess up. At that point, not sure if that's better than just having an indication of whether you've prepared or removed for the turn (which would still be helpful now that I think about it).

Reply:
Okay, I've put in the generic incremented/decremented this turn bits. Actually using them for Unconstrained Fangs I will put in a different pull request since it's separate from Spreading Rot.

The free-form input box would be a new way of doing things. Right now the `change` functions all take deltas from the current value, but this new way would be setting to the provided value. Not saying we shouldn't do it (we should do it if it's what makes playing Unconstrained the most ergonomic), just preparing mentally for the work.

Only semi-related topic, I am sort of opinionated in that I think it behooves the player to keep written records of their energy, prepared beasts, etc. anyway. Too easy to forget whether something has or hasn't yet been done this turn. The site could maybe help this by saving and displaying what the value was on the most recent ready/unready.

Reply:
> The free-form input box would be a new way of doing things. Right now the `change` functions all take deltas from the current value, but this new way would be setting to the provided value. Not saying we shouldn't do it (we should do it if it's what makes playing Unconstrained the most ergonomic), just preparing mentally for the work.

Totally; maybe we start with the simpler option and see how it feels in an actual game. (I'm past due for giving this aspect a go, let alone less experience with Fangs overall.)

> Only semi-related topic, I am sort of opinionated in that I think it behooves the player to keep written records [...] The site could maybe help this by saving and displaying what the value was on the most recent ready/unready.

While I agree that more tracking could be useful, since it is easy to get lost; I see enough people forgetting to Ready/Unready until the next growth when they can't click the gain energy, so I'd be worried that this would possibly add confusion.